src/protocols/bdx/BdxMessages.h

```diff
@@ -28,65 +28,111 @@
 #include <support/BufBound.h>
 #include <support/CodeUtils.h>
 #include <system/SystemPacketBuffer.h>
-
 namespace chip {
-namespace BDX {
+namespace bdx {
+
+enum MessageType : uint8_t
+{
+    kBdxMsg_SendInit      = 0x01,
+    kBdxMsg_SendAccept    = 0x02,
+    kBdxMsg_ReceiveInit   = 0x04,
+    kBdxMsg_ReceiveAccept = 0x05,
+    kBdxMsg_BlockQuery    = 0x10,
+    kBdxMsg_Block         = 0x11,
+    kBdxMsg_BlockEOF      = 0x12,
+    kBdxMsg_BlockAck      = 0x13,
+    kBdxMsg_BlockAckEOF   = 0x14,
+};
+
+enum StatusCode : uint16_t
+{
+    kStatus_None                       = 0x0000,
+    kStatus_Overflow                   = 0x0011,
+    kStatus_LengthTooLarge             = 0x0012,
+    kStatus_LengthTooShort             = 0x0013,
+    kStatus_LengthMismatch             = 0x0014,
+    kStatus_LengthRequired             = 0x0015,
+    kStatus_BadMessageContents         = 0x0016,
+    kStatus_BadBlockCounter            = 0x0017,
+    kStatus_TransferFailedUnknownError = 0x001F,
+    kStatus_ServerBadState             = 0x0020,
+    kStatus_FailureToSend              = 0x0021,
+    kStatus_TransferMethodNotSupported = 0x0050,
+    kStatus_FileDesignatorUnknown      = 0x0051,
+    kStatus_StartOffsetNotSupported    = 0x0052,
+    kStatus_VersionNotSupported        = 0x0053,
+    kStatus_Unknown                    = 0x005F,
+};
 
 enum TransferControlFlags : uint8_t
 {
     // first 4 bits reserved for version
-    kSenderDrive   = (1U << 4),
-    kReceiverDrive = (1U << 5),
-    kAsync         = (1U << 6),
+    kControl_SenderDrive   = (1U << 4),
+    kControl_ReceiverDrive = (1U << 5),
+    kControl_Async         = (1U << 6),
 };
 
 enum RangeControlFlags : uint8_t
 {
-    kDefLen      = (1U),
-    kStartOffset = (1U << 1),
-    kWiderange   = (1U << 4),
+    kRange_DefLen      = (1U),
+    kRange_StartOffset = (1U << 1),
+    kRange_Widerange   = (1U << 4),
 };
 
-/*
- * A structure for representing a SendInit or ReceiveInit message (both contain
- * identical parameters).
+/**
+ * @brief
+ *   Interface for defining methods that apply to all BDX messages.
  */
-struct TransferInit
+struct BdxMessage
```

Comment:
class?

I'd expect struct to be a bit more POD.

[and all the subtypes below]

Reply:
My main reasoning for struct was that it's a heavily data-focused object and it made more sense to me to have default public members instead of getters and setters or explicitly making them public members.

Getters and setters seemed unnecessary for a data structure whose main job is just to organize information to and from message buffers. What would you prefer? Class with public members?